lua/luasnip/nodes/util/snippet_string.lua

```diff
@@ -1,7 +1,30 @@
 local str_util = require("luasnip.util.str")
 local util = require("luasnip.util.util")
 
----@class SnippetString
+---@class LuaSnip.SnippetString.Mark
+---  A kind-of extmark to the text in this buffer.
+---
+---  It moves with inserted text, and has a gravity to control into which
+---  direction it shifts. pos is 1-based and refers to one character in the
+---  string
+---
+---  If the edge is in the middle of multiple characters (for example rgrav=true,
+---  and chars at pos and pos+1 are replaced), the mark is removed.
+---
+---  FIXME(@bew): How is it different than extmarks? (-> why not use them?)
```

Comment:
Completely right, this is an extmark for simple lua-strings (not neovim buffers) :D

Reply:
Updated in 6204bbd3e99526f766bd029773c5f19145b25ef8